scripts/ci-is-vendor-in-sync.sh

```diff
@@ -20,4 +20,6 @@ set -o pipefail
 
 REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
 
-cd $REPO_ROOT && dep status
+cd $REPO_ROOT
```

Comment:
Should this use a make target, e.g. the one added in https://github.com/kubernetes-sigs/cluster-api/pull/692?

Reply:
No, this should check if vendor is in sync not bring vendor in sync. The make target in #692 does the latter